File: WWF/Hands on labs 3.0/Lab10/Completed/Exercise3/ActivityLibrary/PromptActivity.cs
//--------------------------------------------------------------------------------
// This file is a "Sample" as from Windows Workflow Foundation
// Hands on Labs RC
//
// Copyright (c) Microsoft Corporation. All rights reserved.
//
// This source code is intended only as a supplement to Microsoft
// Development Tools and/or on-line documentation.  See these other
// materials for detailed information regarding Microsoft code samples.
// 
// THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
// KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
// IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
// PARTICULAR PURPOSE.
//--------------------------------------------------------------------------------

using System;
using System.ComponentModel;
using System.ComponentModel.Design;
using System.Collections;
using System.Drawing;
using System.Windows.Forms;
using System.Workflow.ComponentModel;
using System.Workflow.ComponentModel.Design;
using System.Workflow.ComponentModel.Compiler;
using System.Workflow.ComponentModel.Serialization;
using System.Workflow.Runtime;
using System.Workflow.Activities;
using System.Workflow.Activities.Rules;

namespace ActivityLibrary
{
    public partial class PromptActivity : System.Workflow.ComponentModel.Activity
	{
        public PromptActivity()
		{
			InitializeComponent();
		}

        private string question;

        public string Question
        {
            get { return question; }
            set { question = value; }
        }

        public static DependencyProperty AnswerProperty = System.Workflow.ComponentModel.DependencyProperty.Register("Answer", typeof(string), typeof(PromptActivity));

        [Description("The answer to the question")]
        [Browsable(false)]
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Visible)]
        public string Answer
        {
            get
            {
                return ((string)(base.GetValue(PromptActivity.AnswerProperty)));
            }
            set
            {
                base.SetValue(PromptActivity.AnswerProperty, value);
            }
        }

        protected override ActivityExecutionStatus Execute(ActivityExecutionContext executionContext)
        {
            PromptForm form = new PromptForm(this.question);
            form.ShowDialog();

            this.Answer = form.Answer;

            return ActivityExecutionStatus.Closed;
        }
	}
}

File: WWF/Hands on labs 3.5/Lab10/VB/after/Exercise2/DesignerHostingApplication/ActivityLibrary/PromptActivity.vb
'--------------------------------------------------------------------------------
' This file is a "Sample" as from Windows Workflow Foundation
' Hands on Labs RC
'
' Copyright (c) Microsoft Corporation. All rights reserved.
'
' This source code is intended only as a supplement to Microsoft
' Development Tools and/or on-line documentation.  See these other
' materials for detailed information regarding Microsoft code samples.
' 
' THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
' KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
' IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
' PARTICULAR PURPOSE.
'--------------------------------------------------------------------------------

Public class PromptActivity
    Inherits Activity

    Private m_question As String
    Public Property Question() As String
        Get
            Return m_question
        End Get
        Set(ByVal value As String)
            m_question = value
        End Set
    End Property

    Public Shared AnswerProperty As DependencyProperty = DependencyProperty.Register("Answer", GetType(String), GetType(PromptActivity))

    <Description("The answer to the question")> _
    <Browsable(False)> _
    <DesignerSerializationVisibility(DesignerSerializationVisibility.Visible)> _
    Public Property Answer() As String
        Get
            Return (CType((MyBase.GetValue(PromptActivity.AnswerProperty)), String))
        End Get
        Set(ByVal Value As String)
            MyBase.SetValue(PromptActivity.AnswerProperty, Value)
        End Set
    End Property

    Protected Overrides Function Execute(ByVal executionContext As ActivityExecutionContext) As ActivityExecutionStatus
        Dim form As New PromptForm(Me.question)
        form.ShowDialog()
        Me.Answer = form.Answer
        Return ActivityExecutionStatus.Closed
    End Function


End Class
